src/librustc_metadata/rmeta/decoder.rs

```diff
@@ -1198,9 +1198,9 @@ impl<'a, 'tcx> CrateMetadata {
         }
     }
 
-    fn get_macro(&self, id: DefIndex) -> MacroDef {
+    fn get_macro(&self, id: DefIndex, sess: &Session) -> MacroDef {
         match self.kind(id) {
-            EntryKind::MacroDef(macro_def) => macro_def.decode(self),
+            EntryKind::MacroDef(macro_def) => macro_def.decode((self, sess)),
```

Comment:
Out of interest -- what caused this to be necessary? Do you know? I can't find a diff in this PR which added the Session parameter to some function...

Reply:
I got an ICE when I just had `.decode(self)`. It arose from `impl<'a, 'tcx> SpecializedDecoder<Span> for DecodeContext<'a, 'tcx> {`. You can grep for `bug!("Cannot decode Span without Session.")`. @eddyb suggested I change this. :)

Reply:
The `decode` method is generic, and in its most basic form doesn't require a `&Session` - perhaps we can make that mandatory?

But the change here is in line with other methods in this file, you'll see what I mean if you look around for a bit.